backup/src/main/java/io/camunda/zeebe/backup/management/InProgressBackupImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.management;
+
+import io.camunda.zeebe.backup.api.BackupIdentifier;
+import io.camunda.zeebe.backup.api.BackupStore;
+import io.camunda.zeebe.scheduler.ConcurrencyControl;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+
+final class InProgressBackupImpl implements InProgressBackup {
+
+  private final BackupIdentifier backupId;
+  private final long checkpointPosition;
+  private final int numberOfPartitions;
+  private final ConcurrencyControl concurrencyControl;
+
+  InProgressBackupImpl(
+      final BackupIdentifier backupId,
+      final long checkpointPosition,
+      final int numberOfPartitions,
+      final ConcurrencyControl concurrencyControl) {
+    this.backupId = backupId;
+    this.checkpointPosition = checkpointPosition;
+    this.numberOfPartitions = numberOfPartitions;
+    this.concurrencyControl = concurrencyControl;
+  }
+
+  @Override
+  public long checkpointId() {
+    return backupId.checkpointId();
+  }
+
+  @Override
+  public long checkpointPosition() {
+    return checkpointPosition;
+  }
+
+  @Override
+  public ActorFuture<Void> findValidSnapshot() {
+    return concurrencyControl.createCompletedFuture();
+  }
+
+  @Override
+  public ActorFuture<Void> reserveSnapshot() {
```

Comment:
:question: What about un-reserving? I guess we'd do that in the close method?

Reply:
Yes. This must be done in `close`.